inode/file_flusher.go

```diff
@@ -346,30 +443,58 @@ func (vS *volumeStruct) doFileInodeDataFlush(fileInode *inMemoryInodeStruct) (er
 	return
 }
 
-func inFlightLogSegmentFlusher(inFlightLogSegment *inFlightLogSegmentStruct) {
+func (vS *volumeStruct) inFlightLogSegmentFlusher(inFlightLogSegment *inFlightLogSegmentStruct) {
 	var (
 		err error
 	)
 
-	// Terminate Chunked PUT
-	err = inFlightLogSegment.Close()
-	if nil != err {
-		err = blunder.AddError(err, blunder.InodeFlushError)
-		inFlightLogSegment.fileInode.Lock()
-		inFlightLogSegment.fileInode.inFlightLogSegmentErrors[inFlightLogSegment.logSegmentNumber] = err
+	inFlightLogSegment.fileInode.Lock()
+	defer inFlightLogSegment.fileInode.Unlock()
+
+	// Handle the race between a DLM-serialized Flush triggering this versus the starvatation condition
+	// doing so... Either one will perform the appropriate steps to enable the Flush() to complete.
+
+	if inFlightLogSegment.Active() {
+		// Terminate Chunked PUT
+		err = inFlightLogSegment.Close()
+		if nil != err {
+			err = blunder.AddError(err, blunder.InodeFlushError)
+			inFlightLogSegment.fileInode.inFlightLogSegmentErrors[inFlightLogSegment.logSegmentNumber] = err
+			delete(inFlightLogSegment.fileInode.inFlightLogSegmentMap, inFlightLogSegment.logSegmentNumber)
+			openLogSegmentLRURemove(inFlightLogSegment)
+			inFlightLogSegment.fileInode.Done()
+			return
+		}
+
+		// Remove us from inFlightLogSegments.logSegmentsMap and let Go's Garbage Collector collect us as soon as we return/exit
 		delete(inFlightLogSegment.fileInode.inFlightLogSegmentMap, inFlightLogSegment.logSegmentNumber)
-		inFlightLogSegment.fileInode.Unlock()
+		openLogSegmentLRURemove(inFlightLogSegment)
+
+		// And we are done
 		inFlightLogSegment.fileInode.Done()
+	}
+}
+
+func chunkedPutConnectionPoolStarvationCallback() {
+	var (
```

Comment:
This is only going to flush 1 inode at a time.   Do we want it to flush multiple inodes until we get below a certain limit or am I missing something?

Reply:
Given the synchronous nature of the call... I see no real benefit in holding up the allocation any longer than necessary. There is no longer any so-called "starvationMode" we transition in and out of that might be expensive in some respects... so I deleted the concept of looping. One less thing to have difficulty tuning, right?

Reply:
Makes sense